packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,401 @@
+
+1. `editor` section is the actual text area to add or update rich text content.
+2. `footer` section consists of `nimble-button` to control each text formatting functionalities like bold, italic, etc, and a
+   `footer-actions` slot element which is typically used to add action buttons to the right bottom of the component.
+
+Example usage of the `nimble-rich-text-editor` in the application layer is as follows:
+
+```html
+<nimble-rich-text-editor>
+    <div slot="footer-actions">
+        <nimble-button>Cancel</nimble-button>
+        <nimble-button>Add Comment</nimble-button>
+    </div>
+</nimble-rich-text-editor>
+```
+
+### API
+
+_Props/Attrs_
+
+-   `markdownValue` - is an accessor used to get and set the markdown value.
+    -   `getter` - this will serialize the content by extracting the Node from the editor and convert it into a markdown string using
+        [prosemirror-markdown serializer](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/to_markdown.ts#L30).
+    -   `setter` - this will parse the markdown string into a Node and load it back into the editor using
+        [prosemirror-markdown parser](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/from_markdown.ts#L199).
+
+_Methods_
+
+-   none
+
+_Events_
+
+-   `input` - event emitted when there is a change in the editor. This can be achieved through Tiptap's [update event](https://tiptap.dev/api/events#update).
+    Below are a few scenarios to understand when this event will fire:
+
+    1. This event will fired for every input in the content of the editor, including text inputs, text formatting changes, and text removals.
+    2. This event will not fire when there are no changes made to the content of the editor. For example, all mouse events, selecting the texts, state
+       changes, etc,
+
+    The emitted event data is in the format of a `CustomEvent` and consists of a `detail` object with an `isEmpty` boolean flag indicating whether the editor
+    is empty or not.
+
+_CSS Classes and CSS Custom Properties that affect the component_
+
+-   The minimum and maximum height of the `editor section` will be fixed value. If the content exceeds the maximum height, we will show the vertical scrollbar to
+    view the content of the editor.
+-   The minimum and maximum width of the `editor section` will be fixed value considering the mobile and desktop view of the component respectively. However, if the
+    screen width is reduced
+    below the specified minimum width, the layout may not handle or adapt to the content.
+-   The `footer section` will be a flexbox container and have the same width as the editor section.
+-   The `formatting option toolbar` in the footer section will be enclosed within a flexbox container, enabling the buttons to wrap and occupy roughly seventy
+    percent of the entire footer. If the buttons exceed the seventy percent limit, they will be positioned below in the same container, aligning from left
```

Comment:
This will be applicable only if we set the maximum height in the component and not in the client. Wrapping the buttons and adding them to the other row will increase the height of the component. So, we thought of having a minimum width for the entire component based on the buttons available in the footer (formatting options + slot elements). For the initial release, we will consider having the four buttons space for the formatting options and the remaining space for the slot as mentioned in the spec.